aiida_vasp/io/poscar.py

```diff
@@ -1,119 +1,176 @@
+        self._parsed_data = {}
+
+    @property
+    def _parsed_object(self):
+        """Return the parsevasp object representing the POSCAR file."""
+
+        try:
+            return Poscar(poscar_dict=aiida_to_parsevasp(self._parsed_data['poscar-structure']), prec=self.precision, conserve_order=True)
+        except SystemExit:
+            return None
+
+    def _parse_file(self, inputs):
+        """Read POSCAR file format."""
+
+        # check if structure have already been loaded, in that case just return
+        if isinstance(self._data_obj, get_data_class('structure')):
+            return {'poscar-structure': self._data_obj}
+
+        # pass file path to parsevasp and try to load file
+        try:
+            poscar = Poscar(file_path=self._data_obj.path, prec=self.precision, conserve_order=True)
+        except SystemExit:
+            self._logger.warning("Parsevasp exited abnormally. " "Returning None.")
+            return {'poscar-structure': None}
+
+        result = parsevasp_to_aiida(poscar)
+
+        return result
 
     def count_kinds(self):
         """
         Count consecutive sites that should use the same potential.
 
         :return: [(kind_name, num), ... ]
         """
-        kind_name_order = [site.kind_name for site in self.structure.sites]
+        kind_name_order = [site.kind_name for site in self._data_obj.sites]
         groups = groupby(kind_name_order)
         counts = [(label, sum(1 for _ in group)) for label, group in groups]
         return counts
 
     @property
     def potentials_order(self):
+        """Get the order of the potentials from the StructureData stored in _data_obj."""
         return [kind[0] for kind in self.count_kinds()]
 
-    def poscar_str(self):
-        """
-        Create a string of the POSCAR contents.
+    def get_string(self):
+        """Create a string of the POSCAR contents."""
 
-        Accounts for lattices which have triple product < 0 by inverting lattice vectors in that case.
-        """
-        cell = np.array(self.structure.cell)
-        if np.linalg.det(cell) < 0:
-            cell = cell * -1
-        comment = self.structure.label or self.structure.get_formula()
-        lattice = '\n'.join([self.LATTICE_ROW_TPL.format(*row, float_fmt=self.float_format) for row in cell])
-        kind_counts = ' '.join([str(count[1]) for count in self.count_kinds()])
-        positions = '\n'.join([
-            self.POS_ROW_TPL.format(*site.position, float_fmt=self.float_format, label=self.structure.get_kind(site.kind_name).symbol)
-            for site in self.structure.sites
-        ])
-        return self.POSCAR_TPL.format(comment=comment, lattice=lattice, kind_counts=kind_counts, positions=positions)
-
-    def write(self, path):
-        destination = py_path.local(path)
-        destination.write(self.poscar_str())
+        lines = self._parsed_object.get_string().split('\n')
+        # Parsevasp replaces the comment line, so we have to update it in case
```

Comment:
Can you either get rid of the commented line if you think it should not be there at all or if it requires some future action to make it work, please add an Issue instead. Assume you will never get around to looking at this again but someone will look at the issues.

Reply:
I removed this and more code around it that is no longer necessary after the element X is introduced. Now the plugin has to determine what potential to use, either automatically, or request input from user. The poscar parser do not anymore care about this, which I think is more natural.